### PR TITLE
Adding Python3.7 wheels support

### PR DESCRIPTION
**Pull request by devin-petersohn**
## What do these changes do?
Build Python 3.7 wheels during travis tests.

## Related issue number
#1115

### CODE REVIEW

**Comment by robertnishihara** [comment]
Nice!

File: python/build-wheel-manylinux1.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ rm -f /usr/bin/python2
 ln -s /opt/python/cp27-cp27m/bin/python2 /usr/bin/python2
 
 mkdir .whl
-for PYTHON in cp27-cp27mu cp34-cp34m cp35-cp35m cp36-cp36m; do
```
Comment (robertnishihara) [inline python/build-wheel-manylinux1.sh]: Whether this works or not just depends on whether the base image already has Python 3.7 on it. The base image is the one used for building Arrow wheels.

**Comment by pcmoritz** [comment]
we should revisit this, arrow now has support for 3.7 I think; see also https://github.com/ray-project/ray/issues/1115

**Review by pcmoritz — APPROVED** [review]
Note: I made sure this works with the earliest numpy compatible with python 3.7 on mac, see changes in the PR.

**Pull request merged**